rivertype/river_type.go

```diff
@@ -111,17 +111,62 @@ type JobRow struct {
 	Tags []string
 }
 
-// JobState is the state of a job. Jobs start as `available` or `scheduled`, and
-// if all goes well eventually transition to `completed` as they're worked.
+// JobState is the state of a job. Jobs start their lifecycle as either
+// JobStateAvailable or JobStateScheduled, and if all goes well, transition to
+// JobStateCompleted after they're worked.
 type JobState string
 
 const (
+	// JobStateAvailable is the state for jobs that are immediately eligible to
+	// be worked.
 	JobStateAvailable JobState = "available"
+
+	// JobStateCancelled is the state for jobs that have been manually cancelled
+	// by user request.
+	//
+	// Cancelled jobs are reaped by the job cleaner service after a configured
+	// amount of time (default 24 hours).
 	JobStateCancelled JobState = "cancelled"
+
+	// JobStateCompleted is the state for jobs that have successfully run to
+	// completion.
+	//
+	// Completed jobs are reaped by the job cleaner service after a configured
+	// amount of time (default 24 hours).
 	JobStateCompleted JobState = "completed"
+
+	// JobStateDiscarded is the state for jobs that have errored enough times
+	// that they're no longer eligible to be retried. Manual user invention
+	// is required for them to be tried again.
+	//
+	// Discarded jobs are reaped by the job cleaner service after a configured
+	// amount of time (default 7 days).
 	JobStateDiscarded JobState = "discarded"
+
+	// JobStateRetryable is the state for jobs that have errored, but will be
+	// retried.
+	//
+	// The job scheduler service changes them to JobStateAvailable when they're
+	// ready to be worked (their `scheduled_at` timestamp comes due).
+	//
+	// Jobs that will be retried very soon in the future may be changed to
+	// JobStateAvailable immediately instead of JobStateRetryable so that they
+	// don't have to wait for the job scheduler to run.
 	JobStateRetryable JobState = "retryable"
-	JobStateRunning   JobState = "running"
+
+	// JobStateRunning are jobs which are actively running.
+	//
+	// If a worker dies unexpectedly while jobs were being worked, jobs will be
```

Comment:
Might be worth clarifying the behavior here as this is specifically true when the client doesn’t have an opportunity to log the failure (program crash, hardware failure, or job that never returns from work).

Reply:
Sure, good idea. Updated!